File: app/src/main/java/com/aiwujie/shengmo/adapter/AtMemberListviewAdapter.java
package com.aiwujie.shengmo.adapter;

import android.content.Context;
import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.aiwujie.shengmo.R;
import com.aiwujie.shengmo.bean.AtGroupMemberData;
import com.aiwujie.shengmo.utils.GlideImgManager;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.aiwujie.shengmo.http.HttpUrl.NetPic;

/**
 * Created by 290243232 on 2016/12/17.
 */
public class AtMemberListviewAdapter extends BaseAdapter{
    private Context context;
    private List<AtGroupMemberData.DataBean> list;
    private LayoutInflater inflater;
    private Handler handler = new Handler();

    public AtMemberListviewAdapter(Context context, List<AtGroupMemberData.DataBean> list) {
        super();
        this.context = context;
        this.list = list;
        inflater = LayoutInflater.from(context);
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.item_listview_atgroup_member, null);
            holder = new ViewHolder(convertView);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }
        AtGroupMemberData.DataBean data = list.get(position);
        if (data.getHead_pic().equals("") || data.getHead_pic().equals(NetPic())) {//"http://59.110.28.150:888/"
            holder.itemAtGroupMemberIcon.setImageResource(R.mipmap.morentouxiang);
        } else {
            GlideImgManager.glideLoader(context, data.getHead_pic(), R.mipmap.morentouxiang, R.mipmap.morentouxiang, holder.itemAtGroupMemberIcon, 0);
        }
        if (!"".equals(data.getCardname())&&null!=data.getCardname()){
            holder.itemAtGroupMemberName.setText(data.getCardname());
        }else {
            holder.itemAtGroupMemberName.setText(data.getNickname());
        }
        return convertView;
    }


    static class ViewHolder {
        @BindView(R.id.item_at_group_member_icon)
        ImageView itemAtGroupMemberIcon;
        @BindView(R.id.item_at_group_member_name)
        TextView itemAtGroupMemberName;

        ViewHolder(View view) {
            ButterKnife.bind(this, view);
        }
    }
}

File: app/src/main/java/com/aiwujie/shengmo/kt/ui/adapter/AtMemberListViewAdapter.kt
package com.aiwujie.shengmo.kt.ui.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.aiwujie.shengmo.R
import com.aiwujie.shengmo.bean.AtGroupMemberData
import com.aiwujie.shengmo.http.HttpUrl.NetPic
import com.aiwujie.shengmo.utils.GlideImgManager

class AtMemberListViewAdapter(var context: Context, var list: List<AtGroupMemberData.DataBean>) : RecyclerView.Adapter<AtMemberListViewAdapter.AtMemberViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): AtMemberListViewAdapter.AtMemberViewHolder {
        return AtMemberListViewAdapter.AtMemberViewHolder(LayoutInflater.from(context).inflate(R.layout.app_item_at_group_member, parent, false));
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: AtMemberListViewAdapter.AtMemberViewHolder, position: Int) {
       var   model :AtGroupMemberData.DataBean= list[position]
        if (""  == model.head_pic    ||  NetPic().equals(model.head_pic) ){
            holder.itemAtGroupMemberIcon.setImageResource(R.mipmap.morentouxiang)
        }else {
            GlideImgManager.glideLoader(context,model.head_pic, R.mipmap.morentouxiang, R.mipmap.morentouxiang, holder.itemAtGroupMemberIcon, 0)
        }

        if ("" !=model.cardname&& model.cardname != null) {
            holder.itemAtGroupMemberName.text = model.cardname
        } else {
            holder.itemAtGroupMemberName.text = model.nickname
        }

        holder.itemView.setOnClickListener {
            AtGroupClick!!.OnClick(position,model.uid,model.nickname,model.cardname,model.head_pic)
        }

    }

    class AtMemberViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var itemAtGroupMemberIcon: ImageView = itemView.findViewById(R.id.item_at_group_member_icon)
        var itemAtGroupMemberName: TextView = itemView.findViewById(R.id.item_at_group_member_name)

    }

   var AtGroupClick :OnAtGroupClick? = null

    interface  OnAtGroupClick {
        fun  OnClick (position:Int,uid:String,nickName:String,cardName:String ,headPic:String)
    }

    fun  setAtGroupClickListener(onAtGroupClick :OnAtGroupClick?) {
        AtGroupClick =onAtGroupClick
    }


}